[RDY]vim-patch:8.0.1554,8.1.1977,8.2.{927,930,932,938,954,963,964,966,980,983,998,999}

janlazo opened the pull request.

janlazo committed:
vim-patch:8.0.1554: custom plugins loaded with --clean

Problem:    Custom plugins loaded with --clean.
Solution:   Do not include the home directory in 'runtimepath'.
https://github.com/vim/vim/commit/072687032683b1994d25a114893d9a6f8bc36612

janlazo commented on src/nvim/option.c:
```diff
@@ -524,11 +524,15 @@ char *get_lib_dir(void)
-  char *const config_home = stdpaths_get_xdg_var(kXDGConfigHome);
+  char *const config_home = clean_arg
+    ? NULL
+    : stdpaths_get_xdg_var(kXDGConfigHome);
```
This removes `stdpath('config')` from `&runtimepath`. Not sure if `stdpath('data')` and other directories should be excluded as well.

jamessan replied:
Yeah, all of the user-specific config should be ignored.  Only the system-wide config should be available.

janlazo commented on runtime/doc/starting.txt:
```diff
@@ -90,6 +90,7 @@ argument.
 --clean		Equivalent to "-u NONE -i NONE":
 		- Skips initializations from files and environment variables.
 		- No 'shada' file is read or written.
+		- Excludes home directory from 'runtimepath'
```
Does this mean `stdpath('config')` directory only?

janlazo committed:
vim-patch:8.2.0927: some sshconfig and ssdhconfig files are not recognized

Problem:    Some sshconfig and ssdhconfig files are not recognized.
Solution:   Add filetype patterns.
https://github.com/vim/vim/commit/da84ac2a6f467d0b9eddcc8709433cda75d16a41

janlazo committed:
vim-patch:8.2.0930: script filetype detection trips over env -S argument

Problem:    Script filetype detection trips over env -S argument.
Solution:   Remove "-S" and "--ignore-environment". (closes vim/vim#5013)
            Add tests.
https://github.com/vim/vim/commit/b5e18f29fac9253b0ccf1fde5e74bff72fa1ba60

janlazo committed:
vim-patch:8.2.0932: missspelling spelllang

Problem:    Missspelling spelllang.
Solution:   Add an "l". (Dominique Pelle)
https://github.com/vim/vim/commit/f154f3ab2c1eea67afc20140c580ad003752dc72

janlazo committed:
eval: fix pvs/v547

janlazo committed:
option: fix pvs/v547

janlazo committed:
vim-patch:8.2.0938: NFA regexp uses tolower ()to compare ignore-case

Problem:    NFA regexp uses tolower() to compare ignore-case. (Thayne McCombs)
Solution:   Use utf_fold() when possible. (ref. neovim vim/vim#12456)
https://github.com/vim/vim/commit/59de417b904bbd204e313f015839317b577bd124

janlazo committed:
vim-patch:8.1.1977: terminal debugger plugin may hang

Problem:    Terminal debugger plugin may hang.
Solution:   Wait longer when still reading symbols.
https://github.com/vim/vim/commit/19c8fe1925f4f7ffa1cc46e64d8bb8b1665ac437

janlazo committed:
vim-patch:8.2.0954: not all desktop files are recognized

Problem:    Not all desktop files are recognized.
Solution:   Add the *.directory pattern. (Eisuke Kawashima, closes vim/vim#3317)
https://github.com/vim/vim/commit/f1f0ff9557353c9a452aac8540411a88cf2da47c

janlazo committed:
vim-patch:8.2.0966: 'shortmess' flag "n" not used in two places

Problem:    'shortmess' flag "n" not used in two places.
Solution:   Make use of the "n" flag consistent. (Nick Jensen, closes vim/vim#6245,
            closes vim/vim#6244)
https://github.com/vim/vim/commit/722e505d1a55dfde5ab62241d10da91d2e10c3c1

janlazo committed:
vim-patch:8.2.0963: number increment/decrement does not work with 'virtualedit'

Problem:    Number increment/decrement does not work with 'virtualedit'.
Solution:   Handle coladd changing. (Christian Brabandt, closes vim/vim#6240,
            closes vim/vim#923)
https://github.com/vim/vim/commit/6c6be9e88d72a60ee279ccad73d018c534b71d66

janlazo committed:
vim-patch:8.2.0964: TextYankPost does not provide info about Visual selection

Problem:    TextYankPost does not provide info about Visual selection.
Solution:   Add the 'visual' key in v:event. (closes vim/vim#6249)
https://github.com/vim/vim/commit/37d1673ce007c46ec2338f3236ae1b7b0324bd25

janlazo committed:
vim-patch:8.2.0980: raku file extension not recognized

Problem:    Raku file extension not recognized. (Steven Penny)
Solution:   Recognize .raku and .rakumod. (closes vim/vim#6255)
https://github.com/vim/vim/commit/a65d8b5bb9e9267c6e4500c67d26c839a64f30eb

janlazo committed:
vim-patch:8.2.0983: SConstruct file type not recognized

Problem:    SConstruct file type not recognized.
Solution:   Use python for SConstruct files. (Roland Hieber)
https://github.com/vim/vim/commit/ad72350360d1a68d251da62a6e8610950926278c

janlazo committed:
vim-patch:8.2.0998: not all tag code is tested

Problem:    Not all tag code is tested.
Solution:   Add a few more test cases. (Yegappan Lakshmanan, closes vim/vim#6284)
https://github.com/vim/vim/commit/3d9207ad2fc98b4f92f77b5a3d52a3a4d25b9561

janlazo committed:
vim-patch:8.2.0999: moving to next sentence gets stuck on quote

Problem:    Moving to next sentence gets stuck on quote.
Solution:   When moving to the next sentence doesn't result in moving, advance
            a character and try again. (closes vim/vim#6291)
https://github.com/vim/vim/commit/2f03e5a0a9cfaabb01c57dd3348a05e86c26623c

teto commented:
went through it and it looked perfect. Thanks for the PVS too. I will be diligent on the next version.c patch, please repost your NA patches ^^''
